subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DefaultDependencyManagementServices.java

```diff
@@ -344,7 +347,15 @@ AttributesSchemaInternal createConfigurationAttributesSchema(InstantiatorFactory
         }
 
         MutableTransformationWorkspaceProvider createTransformerWorkspaceProvider(ProjectLayout projectLayout, ExecutionHistoryStore executionHistoryStore) {
-            return new MutableTransformationWorkspaceProvider(projectLayout, executionHistoryStore);
+            return new MutableTransformationWorkspaceProvider(transformerWorkspaceBaseDirectory(projectLayout), executionHistoryStore);
+        }
+
+        ReservedFileLocation createReservedTransformerWorkspace(ProjectLayout projectLayout) {
+            return new ReservedFileLocation(transformerWorkspaceBaseDirectory(projectLayout));
+        }
+
+        private Provider<Directory> transformerWorkspaceBaseDirectory(ProjectLayout projectLayout) {
+            return projectLayout.getBuildDirectory().dir(".transforms");
         }
```

Comment:
I wish we had named services so this could be an actual service. Now it's too easy to mistake this method for a `create` one.

Making it static would help distinguish it somewhat.

Reply:
An alternative could be to make `MutableTransformationWorkspaceProvider` implement `ReservedFileLocation`. That way we wouldn't need to have this method at all.

Reply:
Good point, `MutableTransformationWorkspaceProvider` now implements `ReservedFileSystemLocation`.